server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -211,6 +212,10 @@ public IndexShard getShard(int shardId) {
         return shards.keySet();
     }
 
+    public Collection<IndexShard> shards() {
+        return shards.values();
+    }
+
```

Comment:
`IndexService` implements `Iterable<IndexShard`, isn't that sufficient for `stopTrackingIndexShards` ?

Reply:
Good point, I missed presence of `iterator()`